### PR TITLE
tools/importer-rest-api-specs: adding a workaround for machine learning registry

### PR DESCRIPTION
**Pull request by wuxu92**
a workaound for https://github.com/Azure/azure-rest-api-specs/issues/25119 also fix https://github.com/hashicorp/pandora/issues/2889

### CODE REVIEW

File: tools/importer-rest-api-specs/components/parser/dataworkarounds/workaround_machinelearning_25142.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+func (workaroundMachineLearning25142) Name() string {
+	return "MachineLearningServices / 25142"
+}
+
+func (workaroundMachineLearning25142) Process(apiDefinition models.AzureApiDefinition) (*models.AzureApiDefinition, error) {
+	resource, ok := apiDefinition.Resources["RegistryManagement"]
+	if !ok {
+		return nil, fmt.Errorf("couldn't find API Resource MachineLearningServices")
```
Comment (tombuildsstuff) [inline tools/importer-rest-api-specs/components/parser/dataworkarounds/workaround_machinelearning_25142.go:30]:
```suggestion
		return nil, fmt.Errorf("couldn't find API Resource RegistryManagement")
```

File: tools/importer-rest-api-specs/components/parser/dataworkarounds/workarounds.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ var workarounds = []workaround{
 
 	// @tombuildsstuff: we also have to account for package names which aren't valid in Go:
 	workaroundInvalidGoPackageNames{},
+
+	workaroundMachineLearning25142{},
```
Comment (tombuildsstuff) [inline tools/importer-rest-api-specs/components/parser/dataworkarounds/workarounds.go:27]: Can we put this up the top with the other service fixes? These two at the bottom are kinda special-case catch-all's

**Review by tombuildsstuff — COMMENTED** [review]
A couple of minor comments, but this otherwise LGTM 👍

File: tools/importer-rest-api-specs/components/parser/dataworkarounds/workaround_machinelearning_25142.go (hunk 1 above)
Comment (wuxu92) [reply tools/importer-rest-api-specs/components/parser/dataworkarounds/workaround_machinelearning_25142.go]: updated

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by wuxu92** [comment]
@tombuildsstuff just to confirm, will the azure-go-sdk be updated automatically with the new workaround, or any other actions needed?

**Comment by tombuildsstuff** [comment]
@wuxu92 the API Definitions get regenerated to include the new status code (#2897) - once that's merged the Go SDK gets regenerated - once that's merged it's tagged/released

**Pull request merged**